src/box/lua/load_cfg.lua

```diff
@@ -394,9 +396,17 @@ local function normalize_uri_list_for_replication(port_list)
     return {port_list}
 end
 
+local function normalize_tt_hostname(name)
+    if name == nil then
+        return nil
+    end
+    return string.lower(name)
```

Comment:
Why do we need to "normalize" the name? IMO it'd be unexpected to the user that the actual cluster name differs from the value passed to `box.cfg`.

Reply:
Because it was proposed by the product manager and approved in the RFC. Personally though I agree that the auto-lowering shouldn't be done. Better raise an error if the name has uppercase letters. If you want to push your version, you need to talk to the product manager (Mons).

Reply:
Hmm, I must have missed that. I can't find this in the approved [RFC](https://www.notion.so/tarantool/Instance-replicaset-and-cluster-names-in-addition-to-UUIDs-9031bb4301eb409dadb86fc35065332a). Can you point me at it?

Reply:
Oh no, sorry, I now found it in my VK Teams chat with Mons. He chose the auto-lowering and I forgot to add it to the RFC :(

Reply:
Well, since there are several developers, which are in doubt about this decision, it worth to ask a reason of the decision.

Reply:
Deal with @Mons please. I will do whatever is decided.

Reply:
That's your responsibility: if there are questions to your code, you should reach anybody you lean in your decisions and provide answers.

https://github.com/tarantool/tarantool/wiki/User:Totktonada#youre-the-owner-now

Reply:
Ok, reaching. @Mons, tell me how should I explain this in the code. Because I have no idea what reasoning is behind the autolowering.

Reply:
Dns-like names are case-insensitive.
The canonical form for dns-like name is lowercase.
We could leave names as it was passed, but I think that will lead to user mistakes or implementation errors.

Personally, I don't care. You may leave names as is, but they must behave in a case insensitive way (so that one may change the case for name from run to run)

Reply:
Is it to ease transition from 7 bit encodings to modern 8 bit ones? If so, it looks irrelevant for several decades.

BTW, RFC 4343 (DNS Case Insensitivity Clarification) suggests to output the original case, but compare case-insensitively. At least it looks better than silent unexpected transformation.

The transformation is unexpected at least because those names do not look like DNS names.

And the case-insensitive comparisons are unexpected too due to the same reason.

Since it is not DNS-like, I see no sense to refer to DNS when we're talking about user's expectations.

I can't accept your argument. (OTOH, you don't have to convince me.)

Reply:
Since it is well-documented with explicit explanation what is done and why - both is documentation and in sources - I consider it resolved as 'requirement compliant' and do not start overturn the requirement at this point.